packages/caliper-core/test/worker/tx-observers/internal-tx-observer.js

```diff
@@ -0,0 +1,218 @@
+
+            await observer.activate(1, 'round1');
+
+            // Fast forward the clock to simulate intervals
+            clock.tick(1000);
+            expect(messengerMock.send).to.have.been.calledOnce;
+
+            clock.tick(1000);
+            expect(messengerMock.send).to.have.been.calledTwice;
+
+            sinon.assert.notCalled(warnLogger);
+            sinon.assert.notCalled(errorLogger);
+        });
+
+        it('should log warning if interval is invalid and use default value', async () => {
+            sinon.stub(ConfigUtil, 'get').returns(-1); // Invalid interval
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+
+            await observer.activate(1, 'round1');
+
+            // Check if default interval was used
+            expect(observer.updateInterval).to.equal(1000);
+            sinon.assert.calledOnce(warnLogger);
+        });
+    });
+
+    describe('Deactivating and stopping updates', () => {
+        it('should deactivate the observer, stop updates and send final messages', async () => {
+            sinon.stub(ConfigUtil, 'get').returns(1000);
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+
+            await observer.activate(1, 'round1');
+            clock.tick(1000);
+
+            await observer.deactivate();
+            expect(observer.intervalObject).to.be.null;
+
+            expect(messengerMock.send).to.have.been.calledThrice; // Once during activate, twice during deactivate
+
+            // Ensure the first two messages were TxUpdateMessages
+            const firstCallArg = messengerMock.send.getCall(0).args[0];
+            const secondCallArg = messengerMock.send.getCall(1).args[0];
+            expect(firstCallArg).to.be.instanceOf(TxUpdateMessage);
+            expect(secondCallArg).to.be.instanceOf(TxUpdateMessage);
+
+            // Ensure the third message is a TxResetMessage
+            const txResetMessage = new TxResetMessage('mocked-messenger-uuid', [managerUuid]);
+            sinon.assert.calledWith(messengerMock.send, txResetMessage);
+
+            // Ensure the error logger was not called
+            sinon.assert.notCalled(errorLogger);
+        });
+
+
+        it('should log error if deactivation fails due to message sending', async () => {
+            sinon.stub(ConfigUtil, 'get').returns(1000);
+            messengerMock.send.rejects(new Error('Send error'));
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+
+            await observer.activate(1, 'round1');
+            clock.tick(1000);
+
+            await expect(observer.deactivate()).to.be.rejectedWith('Send error');
+            sinon.assert.calledOnce(errorLogger);
+        });
+    });
+
+    it('should send the correct update message with the correct transaction count when activated', async () => {
```

Comment:
This is testing that when the observer sends a message at an interval point that the correct number of transactions submitted is included in the message. The above test implies this is something that happens at activation time. So we should change the test to be more relevant (ie it's not related to activation)